File: Apache_Simulator/Source/Apache_Simulator/Switches/EngineSwitch.cpp
// Fill out your copyright notice in the Description page of Project Settings.


#include "EngineSwitch.h"

// Sets default values for this component's properties
UEngineSwitch::UEngineSwitch()
{
	// Set this component to be initialized when the game starts, and to be ticked every frame.  You can turn these features
	// off to improve performance if you don't need them.
	PrimaryComponentTick.bCanEverTick = true;

	// ...
}


// Called when the game starts
void UEngineSwitch::BeginPlay()
{
	Super::BeginPlay();
	IsOn = false;
	// ...
	
}


// Called every frame
void UEngineSwitch::TickComponent(float DeltaTime, ELevelTick TickType, FActorComponentTickFunction* ThisTickFunction)
{
	Super::TickComponent(DeltaTime, TickType, ThisTickFunction);

	if (IsOn)
		ActionWhenOn();
	else
		ActionWhenOff();
	// ...
}

void UEngineSwitch::ActionWhenOn()
{
	ApacheEngine->IsOn = true;

}

void UEngineSwitch::ActionWhenOff()
{
	ApacheEngine->IsOn = false;
}

File: ApacheSim/Assets/Scripts/Switches/EngineSwitch.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class EngineSwitch : B_Switch
{
    public ApacheEngine ApacheEngine;
    public override void ActionWhenOff()
    {
        ApacheEngine.IsOn = false;
    }

    public override void ActionWhenOn()
    {
        ApacheEngine.IsOn = true;
    }

    // Start is called before the first frame update
    void Start()
    {
        IsOn = false;
    }

    // Update is called once per frame
    void Update()
    {

        if (IsOn)
            ActionWhenOn();
        else
            ActionWhenOff();

    }
}

